src/main/java/org/bananarama/BananaRama.java

```diff
@@ -42,10 +42,18 @@
     
     private final StripedLock slock;
     private final Map<Class<?>, Adapter<?>> adapters;
-    private static final Map<Class<?>, Class<? extends Adapter>> customAdapters = new ConcurrentHashMap<>();
+    private final Map<Class<?>, Class<? extends Adapter>> customAdapters = new ConcurrentHashMap<>();
     
-    public static void registerAdapter(Class<? extends Adapter> adapter, Class forEntiy) {
+    /**
+     * Map the specified {@link Adapter} for a given entity. 
+     * This method is usefull to allow final class from external libraries to be managed by BananaRama
+     * @param adapter
+     * @param forEntiy 
+     * @return this instance
+     */
+    public BananaRama registerAdapter(Class<? extends Adapter> adapter, Class forEntiy) {
```

Comment:
Ok better, but again, same problem, this implies that annotated class are not necessarily loaded as expected which is very dangerous.